IBPSA/Fluid/Geothermal/Aquifer/MultiWell.mo

```diff
@@ -0,0 +1,428 @@
+     "Undisturbed ground temperature (warm well)" annotation (
+      Dialog(group="Properties of ground"));
+  parameter IBPSA.Fluid.Geothermal.Aquifer.Data.Template aquDat
+     "Aquifer thermal properties" annotation (choicesAllMatching=true);
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal "Nominal mass flow rate" annotation (
+      Dialog(group="Hydraulic circuit"));
+  parameter Modelica.Units.SI.PressureDifference dpAquifer_nominal(displayUnit=
+        "Pa") "Pressure drop at nominal mass flow rate in the aquifer"
+                                                             annotation (
+      Dialog(group="Hydraulic circuit"));
+  parameter Modelica.Units.SI.PressureDifference dpWell_nominal(displayUnit="Pa")
+    "Pressure drop at nominal mass flow rate in the well" annotation (
+      Dialog(group="Hydraulic circuit"));
+  parameter Modelica.Units.SI.PressureDifference dpExt_nominal(displayUnit="Pa")
+    "Pressure drop at nominal mass flow rate in the above-surface system (used to size the head of the well pump)" annotation (
+      Dialog(group="Hydraulic circuit"));
+
+  Modelica.Blocks.Interfaces.RealInput u(
+      final min=-1,
+      final max=1,
+      final unit="1")
+      "Pump control input (-1: extract from hot well, +1: extract from cold well, 0: off)"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+        iconTransformation(extent={{-140,40},{-100,80}})));
+
+  Modelica.Fluid.Interfaces.FluidPort_a port_Col(
+    redeclare final package Medium = Medium) "Fluid connector" annotation (Placement(transformation(extent={
+            {-70,90},{-50,110}}), iconTransformation(extent={{-70,90},{-50,110}})));
+
+  Modelica.Fluid.Interfaces.FluidPort_a port_Hot(
+    redeclare final package Medium = Medium) "Fluid connector" annotation (Placement(transformation(extent={
+            {50,90},{70,110}}), iconTransformation(extent={{50,90},{70,110}})));
+
+  Movers.Preconfigured.SpeedControlled_y pumCol(
+    redeclare final package Medium = Medium,
+    addPowerToMedium=false,
+    final m_flow_nominal=m_flow_nominal,
+    final dp_nominal=dpAquifer_nominal*2 + dpWell_nominal*2 + dpExt_nominal)
+    "Pump to extract from cold well" annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=90,
+        origin={-80,20})));
+  Movers.Preconfigured.SpeedControlled_y pumHot(
+    redeclare final package Medium = Medium,
+    addPowerToMedium=false,
+    final m_flow_nominal=m_flow_nominal,
+    final dp_nominal=dpAquifer_nominal*2 + dpWell_nominal*2 + dpExt_nominal)
+    "Pump to extract from hot well" annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={80,20})));
+
+  IBPSA.Fluid.MixingVolumes.MixingVolume volCoo[nVol](
+    redeclare final package Medium = Medium,
+    each final T_start=TCoo_start,
+    each final m_flow_nominal=m_flow_nominal,
```

Comment:
I suggest to make all these components below protected as a user typically does not need to see these result. If so, we can make a temperature vector `TCoo` and `THot` that has the temperatures of the control volumes so a user can look at them. All other results I doubt a user needs to see (and if so, one could simulate the model with reporting protected variables).

Reply:
Do you mean to make below protected only the `MixingVolume` components or also `HeatCapacitor` and `ThermalResistor` components?

Reply:
Discuss yesterday at the meeting. I made below protected volumes, capacitors and resistances and I created two temperature arrays `TAquHot` and `TAquCol` to output the temperatures in the aquifer. But I removed from protected `rC` as I think that users should be able to see at which distance these temperature are calculated.

Reply:
I just realized that (obviously), by removing `rC` from protected, this appears in the parameter dialog box, and it shouldn't. Therefore, I created a new variable `rVol`, which is simply equal to `rC`, but it is accessible to users. The alternative would be to have `rC` directly as variable, but then I don't think we can have it under `initial equation`